src/Seq.Input.HealthCheck/HttpHealthCheck.cs

```diff
@@ -40,15 +43,32 @@ class HttpHealthCheck
         const int InitialContentChars = 16;
         const string OutcomeSucceeded = "succeeded", OutcomeFailed = "failed";
 
-        public HttpHealthCheck(HttpClient httpClient, string title, string targetUrl, JsonDataExtractor extractor, bool bypassHttpCaching)
+        public HttpHealthCheck(HttpClient httpClient, string title, string targetUrl, string authenticationHeaderValue, string optionalHeader, JsonDataExtractor extractor, bool bypassHttpCaching)
         {
             _httpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
             _title = title ?? throw new ArgumentNullException(nameof(title));
             _targetUrl = targetUrl ?? throw new ArgumentNullException(nameof(targetUrl));
+            _authenticationHeaderValue = authenticationHeaderValue;
+            _optionalHeader = optionalHeader;
             _extractor = extractor;
             _bypassHttpCaching = bypassHttpCaching;
         }
 
+        private static void AddOrOverwriteHeader(HttpRequestMessage request, string attribute, string value)
+        {
+            // opportunistic abort: header values cannot be empty (https://datatracker.ietf.org/doc/html/rfc7230#section-3.2)
+            if (string.IsNullOrWhiteSpace(value))
+            {
+                return;
+            }
+            // the api does not allow overwriting, therefore removal has to happen first.
+            if (request.Headers.Contains(attribute))
```

Comment:
I think duplicates are in fact allowed in request headers (horrible as that is), so we should probably just pass through each value in turn.

Reply:
The request API does not allow duplicate header fields, causing an exception as it is a dictionary with an existing key: 
![image](https://user-images.githubusercontent.com/6331542/158959143-114d69bc-cbaa-4a9d-9f94-035820d2443a.png)